src/Loader/CucumberNDJsonAstLoader.php

```diff
@@ -157,6 +158,15 @@ private static function getTables(array $items): array
             static function ($tableJson) {
                 $table = [];
 
+                if (empty($tableJson['tableHeader']) && !empty($tableJson['tableBody'])) {
+                    throw new NodeException(
+                        sprintf(
+                            'Table header is required when a table body is provided for the example on line %s.',
+                            $tableJson['location']['line'],
+                        )
+                    );
+                }
+
                 $table[$tableJson['tableHeader']['location']['line']] = array_column($tableJson['tableHeader']['cells'], 'value');
```

Comment:
I might be missing something, but it looks like now we're not testing/handling the case that originally prompted this whole thing, where `tableHeader` isn't present and `tableBody` is empty so there is no ExampleTableNode to load?

Don't we need to array_filter the `$items` - either before or after the array_map - to skip over any that are like that?

Reply:
But `$items` represents multiple tables. But now that I think about it, the following condition should be clearer than the current one:
```php
                if (!isset($tableJson['tableHeader']) && $tableJson['tableBody'] === []) {
```

PS: There's a test that checks that the exception is thrown.

Let me know if anything is still unclear.

Reply:
Yeah, but I'm talking about this case https://github.com/Behat/Gherkin/issues/286#issuecomment-2849191554 where $items contains a table that has no `tableHeader` *and* no `tableBody` (the case that explains why `tableHeader` is not required in the schema).

As it stands, that will pass the check, then fail on the `tableHeader` key being undefined.

We just need to ignore examples like that, which come from gherkin like [this](https://github.com/cucumber/gherkin/blob/37f1871eb1a1350c40fe19beda5d28b083eeb976/testdata/good/incomplete_scenario_outline.feature#L13)

Reply:
I see your point, as I understand the code is also wrong and we're missing a test case.

It should be something like:
```text
if (body is empty) {       // header is not important at this point
  return
}

if (header is not set) {   // here it is implied that body is not empty
  throw exception
}

load header and body
```
correct?

Reply:
Yep, perfect thanks, and then just wrap that in an array_filter to strip out any where it returned early.

It should pass with the NDJSON in that comment I linked.